templates.go

```diff
@@ -0,0 +1,181 @@
+		Method: "GET",
+		Path:   "/templates/" + id,
+		Target: tmpl,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return tmpl, nil
+}
+
+// TemplateResp defines the response after template manipulation
+type TemplateResp struct {
+	TemplateID int64 `json:"TemplateId"`
+	Name       string
+	Active     bool
+
+	ErrorCode int
+	Message   string
+}
+
+func (t *templates) Create(ctx context.Context, tmpl *Template) (*TemplateResp, error) {
+	tmplResp := new(TemplateResp)
+	_, err := t.pm.Exec(ctx, &Request{
+		Method:  "POST",
+		Path:    "/templates/",
+		Payload: tmpl,
+		Target:  tmplResp,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return tmplResp, nil
+}
+
+func (t *templates) Edit(ctx context.Context, id string, tmpl *Template) (*TemplateResp, error) {
+	tmplResp := new(TemplateResp)
+	_, err := t.pm.Exec(ctx, &Request{
+		Method:  "PUT",
+		Path:    "/templates/" + id,
+		Payload: tmpl,
+		Target:  tmplResp,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return tmplResp, nil
+}
+
+// TemplateList defines a list of template entities
+type TemplateList struct {
+	TemplateCount int64
+	Templates     []*Template
+}
+
+func (t *templates) List(ctx context.Context, count, offset int) (*TemplateList, error) {
+	urlBuilder := url.URL{
+		Path: "/templates",
+		RawQuery: url.Values{
+			"count":  {strconv.Itoa(count)},
+			"offset": {strconv.Itoa(offset)},
+		}.Encode(),
+	}
+
+	tmplList := new(TemplateList)
+	_, err := t.pm.Exec(ctx, &Request{
+		Method: "GET",
+		Path:   urlBuilder.String(),
+		Target: tmplList,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return tmplList, nil
+}
+
+func (t *templates) Delete(ctx context.Context, id string) (*TemplateResp, error) {
+	tmplResp := new(TemplateResp)
+	_, err := t.pm.Exec(ctx, &Request{
+		Method: "DELETE",
+		Path:   "/templates/" + id,
+		Target: tmplResp,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return tmplResp, nil
+}
+
+// TemplateValidation defines the structure of a template sent for validation with the API
+type TemplateValidation struct {
+	Subject                    string
+	HTMLBody                   string `json:"HtmlBody"`
+	TextBody                   string
+	TestRenderModel            map[string]interface{}
+	InlineCSSForHTMLTestRender bool `json:"InlineCssForHtmlTestRender"`
+}
+
+// TemplateValidationResp defines the result of a template validation call
+type TemplateValidationResp struct {
+	AllContentIsValid      bool
+	ContentIsValid         bool
+	ValidationErrors       []TemplateValidationErr
+	RenderedContent        string
+	SuggestedTemplateModel map[string]interface{}
+}
+
+// TemplateValidationErr defines an error that occured while validating a template
+type TemplateValidationErr struct {
+	Message           string
+	Line              int
+	CharacterPosition int
+}
+
+func (t *templates) Validate(ctx context.Context, tmpl *TemplateValidation) (*TemplateValidationResp, error) {
+	tmplResp := new(TemplateValidationResp)
+	_, err := t.pm.Exec(ctx, &Request{
```

Comment:
Yeah, I think the http status code indicated whether or not they were able to perform the validation, rather than the result of the validation itself.